partner_multi_company/models/res_company.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2025 Simone Rubino - PyTech
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    @api.model_create_multi
+    def create(self, vals_list):
```

Comment:
It's also a good habit to document why are you overriding the method:

```suggestion
    def create(self, vals_list):
        # Inject the created company into the allowed companies of users
        # already indicated in the company
```

Reply:
Right, done, please check